iron-label.html

```diff
@@ -0,0 +1,141 @@
+
+        <iron-label>
+          Label for the Button
+          <paper-button>button</paper-button>
+        </iron-label>
+
+2. place some elements inside iron-label and target one with the `iron-label-target` attribute.
+The other elements will provide the label for that element
+Note: This is not necessary if the element you want to label is the first
+element child of iron-label:
+
+        <iron-label>
+          <span>Label for the Button</span>
+          <paper-button iron-label-target>button</paper-button>
+        </iron-label>
+
+        <iron-label>
+          <paper-button>button</paper-button>
+          <span>Label for the Button</span>
+        </iron-label>
+
+3. Set the `for` attribute on the `iron-label` element with the id of the target
+element in the same scope:
+
+        <iron-label for="foo">
+          Context for the button with the "foo" class"
+        </iron-label>
+        <paper-button id="foo">Far away button</paper-button>
+
+All taps on the `iron-label` will be forwarded to the "target" element.
+
+@group Iron Elements
+@element iron-label
+@demo demo/index.html
+@homepage polymer.github.io
+-->
+<link rel="import" href="../polymer/polymer.html">
+
+<dom-module id="iron-label">
+  <script>
+    (function() {
+
+      // generate a unique id for this element
+      var ID = 0;
+      function generate(node) {
+        var id = node.id;
+        if (!id) {
+          id = 'iron-label-' + ID++;
+          Polymer.dom(node).setAttribute('id', id);
+        }
+      }
+
+      Polymer({
+        is: 'iron-label',
+
+        listeners: {
+          'tap': '_tapHandler'
+        },
+
+        properties: {
+          /**
+           * An ID reference to another element that needs to be
+           * labelled by this `iron-label` element.
+           */
+          for: {
+            type: String,
+            reflectToAttribute: true,
+            observer: '_forChanged'
+          }
+        },
+
+        created: function() {
+          generate(this);
+          this._forElement = null;
+        },
+
+        get _target() {
+          if (this.for) {
+            // use explicit target
+            var scope = Polymer.dom(this).getOwnerRoot();
+            return Polymer.dom(scope).querySelector('#' + this.for);
+          } else {
+            // use implicit target
+            var el = Polymer.dom(this).querySelector('[iron-label-target]');
+            if (!el) {
+              el = Polymer.dom(this).firstElementChild;
+            }
+            return el;
+          }
+        },
+
+        attached: function() {
+          this._forChanged();
+        },
+
+        _tapHandler: function(ev) {
+          if (!this._forElement) {
+            return;
+          }
+          var target = Polymer.dom(ev).localTarget;
+          if (target === this._forElement) {
+            return;
+          }
+          this._forElement.focus();
+          this._forElement.click();
+        },
+
+        _tie: function() {
```

Comment:
I think I get what this does, looking at the implementation, but as an implementor of overriding behavior, the naming here might confuse me.